src/main/java/TARS/ui/TARS.java

```diff
@@ -0,0 +1,559 @@
+            \tHow can I help you today?""";
+    public static final String GOODBYE_MESSAGE = "\tGoodnight Captain. Sleep well.";
+
+
+    private static ArrayList<Task> taskList = new ArrayList<Task>();
+
+    private static final String dataFolderPath = System.getProperty("user.dir") + "/data";
+    private static final String dataFilePath = dataFolderPath + "/tasks.txt";
+
+    /**
+     * The main entry point into TARS Chatbot Application
+
+     * This method initializes the chatbot, displays a welcome message
+     * and maintains a handler loop for user input until "bye" command is entered.
+     * Users can add tasks, list tasks, mark tasks as done, unmark tasks
+
+     * Commands supported
+     * - Add Task: Any text input added as task (Exceptions below)
+     * - List Tasks: command "list"
+     * - Marking Tasks as done: command "mark {task_number}"
+     * - Marking Tasks as undone: command "unmark {task_number}"
+     * - Exit: command "bye"
+     *
+     * @param args String buffer
+     */
+    public static void main(String[] args) {
+        printHelloMessage();
+        ArrayList<String> fileData = loadFile();
+        initializeTaskList(fileData);
+//        for (int i = 0; i < 4; i++) {
+//            System.out.println(fileData[i]);
+//        }
+
+        String line;
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            line = sc.nextLine().strip();
+
+            try {
+                CommandType commandType = parseCommandType(line);
+
+                if (commandType == CommandType.BYE) {
+                    break;
+                }
+
+                String commandBody = parseCommandBody(line, commandType);
+                commandHandler(commandType, commandBody);
+                writeFile();
+
+            } catch (Exception e) {
+                printResponseMessage(e.getMessage());
+            }
+
+        }
+
+        printGoodbyeMessage();
+        sc.close();
+    }
+
+    private static void printResponseMessage(String message) {
+        System.out.println(LINE_SEPERATOR);
+        System.out.println(message);
+        System.out.println(LINE_SEPERATOR);
+    }
+
+    /**
+     * Prints the hello message to welcome the user.
+     */
+    private static void printHelloMessage() {
+        printResponseMessage(HELLO_MESSAGE);
+    }
+
+    /**
+     * Prints the goodbye message before exiting.
+     */
+    private static void printGoodbyeMessage() {
+        printResponseMessage(GOODBYE_MESSAGE);
+    }
+
+    /**
+     * Parses user input line for commandType
+     * Validation throws exception for empty strings, returns TARS.command.CommandType.INVALID for unknown commands
+     * @param line user input String as a command
+     * @return first word in line as commandType
+     */
+    private static CommandType parseCommandType(String line) throws IndexOutOfBoundsException, IllegalArgumentException {
+        // Check for empty line
```

Comment:
This comment is not right before the relevant line, furthermore, you do not need to write comments for things that are obvious through the code.